akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -628,6 +639,22 @@ object ByteString {
       }
     }
 
+    override def copyToArray[B >: Byte](xs: Array[B], start: Int, len: Int): Int = {
+      if (isCompact) bytestrings.head.copyToArray(xs, start, len)
+      else {
+        val bsIterator = bytestrings.iterator
+        var pos = 0
+        // min of the bytes available top copy, bytes there is room for in dest and the requested number of bytes
+        val bytesToCopy = len.min(xs.length - start)
+        while (pos < bytesToCopy) {
+          val current = bsIterator.next()
```

Comment:
`bsIterator` could run out here, right? since despite the comment, we don't first look at how many bytes there are available to copy

Reply:
Ah, also leftover from the old internal impl.  we should cap with length as well.